include/deal.II/base/patterns.h

```diff
@@ -2015,6 +2051,19 @@ namespace Patterns
       }
     };
 
+    // Utility function with default Pattern
+    template<typename T>
+    std::string to_string(const T &t)
+    {
+      return Convert<T>::to_string(t);
+    }
+
+    // Utility function with default Pattern
+    template<typename T>
+    void to_value(const std::string &s, T &t)
+    {
+      t = Convert<T>::to_value(s);
```

Comment:
I would still prefer to write 
```
auto t=to_value<int>("42");
```
to 
```
int t;
to_value("42", t)
```

Reply:
This defeats the purpose, especially for complex types. I find this orrible to read:

~~~cpp
std::tuple<std::vector<unsigned int>, std::map<types::manifold_id, double> > t;

t = to_value<std::tuple<std::vector<unsigned int>, std::map<types::manifold_id, double> > >(some_string);
~~~
of course, one could use
~~~
t = to_value<decltype(t)>(some_string);
~~~
but again, I find it orrible (but better than above!).

Reply:
I'm against this solution, as one could easily be mistaken by writing 
~~~
to_value("abc", t);
~~~
ignoring the return type of the function. This would be a really difficult bug to find.

Reply:
Well, that's no different than for any other function with a return value.

But you're the author, and I think we ought to let you have the say in design decisions like this. So I'm fine with it!